solver/src/driver.rs

```diff
@@ -1,7 +1,47 @@
-use crate::ethereum::SettlementContract;
+use crate::{ethereum::SettlementContract, naive_solver, orderbook::OrderBookApi};
+use anyhow::{Context, Result};
 use std::{sync::Arc, time::Duration};
 
+const SETTLE_INTERVAL: Duration = Duration::from_secs(30);
+
 pub struct Driver {
     contract: Arc<dyn SettlementContract>,
-    max_order_age: Duration,
+    orderbook: OrderBookApi,
+}
+
+impl Driver {
+    async fn run_forever(&mut self) {
+        loop {
+            println!("starting settle attempt");
+            match self.single_run().await {
+                Ok(()) => println!("ok"),
+                Err(err) => println!("error: {:?}", err),
+            }
+            tokio::time::delay_for(SETTLE_INTERVAL).await;
+        }
+    }
```

Comment:
Here as well I'd slightly prefer a scheduler, but less important than in the orderbook (this class here is still small and relatively focused on actually doing the runloop) so it can also wait.